Libraries/Image/RCTImageView.m

```diff
@@ -194,7 +194,18 @@ - (void)reloadImage
         }
       } else {
         if (_onLoad) {
-          _onLoad(nil);
+          if (dimensions != nil) {
+            CGSize size = [dimensions CGSizeValue];
```

Comment:
Could we greatly simplify this PR by using `image.size` (may need to be careful to account for screen scale -- but it sure would make this PR super short :D )?